pkg/controller/unifiedpushserver/util.go

```diff
@@ -0,0 +1,55 @@
+package unifiedpushserver
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	aerogearv1alpha1 "github.com/aerogear/unifiedpush-operator/pkg/apis/aerogear/v1alpha1"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type image struct {
+	defaultImage string
+	envVarName   string
+}
+
+func (i image) image() string {
+	customImage, found := os.LookupEnv(i.envVarName)
+	if found {
+		return customImage
+	}
+	return i.defaultImage
+}
+
+var (
+	proxy       = image{"docker.io/openshift/oauth-proxy:v1.1.0", "OAUTH_PROXY_IMAGE"}
```

Comment:
IMHO It all should be a spec of the CR since are values that could be need to change and indeed we could change the config in the CR without the need of a new build or code/changes.

Reply:
NOTE: If these ENV VARs are used for all Kind/CRDs, implemented and managed for the operator then could be an ENV VAR of the operator defined in the operator.yaml

Reply:
Going to mark this one as resolved as I think it's really the same case as the other comment above at https://github.com/aerogear/unifiedpush-operator/pull/6/files#r282880591

Thanks!